calico/openstack/mech_calico.py

```diff
@@ -675,6 +683,52 @@ def resync_endpoints(self, context):
                 )
                 self.transport.endpoint_created(port)
 
+        # Finally, scan each of the ports in changes_ports. Work out if there
+        # are any differences. If there are, write out to etcd.
+        changed_endpoints = (e for e in endpoints if e.id in changes_ports)
+
+        for endpoint in changed_endpoints:
+            # Get the endpoint data from etcd.
+            try:
+                endpoint = self.transport.get_endpoint_data(endpoint)
+            except etcd.EtcdKeyNotFound:
+                # The endpoint is gone. That's fine.
+                LOG.info("Failed to update deleted endpoint %s", endpoint.id)
+                continue
+
+            with context.session.begin(subtransactions=True):
+                port = self.db.get_port(context, endpoint.id)
+
+            # Get the data for both.
+            try:
+                etcd_data = json.loads(endpoint.data)
+            except Exception:
```

Comment:
Would `ValueError` and `TypeError` be enough here?  I _think_ that's all loads can raise.